TYPO3.Flow/Classes/TYPO3/Flow/Http/Request.php

```diff
@@ -373,6 +417,136 @@ public function getContent($asResource = false)
     }
 
     /**
+     * Set a trusted proxy header.
+     *
+     * This is primarily useful for testing. If you want to set trusted headers, use the configuration setting
+     * "TYPO3.Flow.http.trustedProxies.headers" instead.
+     *
+     * @param string $type One of the HEADER_* constants
+     * @param string $name The header name to trust or an empty string to not trust this header
+     * @return void
+     */
+    public static function setTrustedProxyHeader($type, $name)
```

Comment:
Well, actually it's supposed to only overwrite the already configured headers (say you want to trust the "Client-Ip" header instead for this single request). In testing it's ofc used to directly set/add the headers, as they are not loaded from configuration there. Not sure if there's really a plausible use case outside of testing, but it could be.